### PR TITLE
T8/llenar ingresos

### PR DESCRIPTION
**Pull request by Ral15**
#### What's this PR do?
This PR has the members section completed, as well as some parts of the income, but IGNORE that :)
#### Where should the reviewer start?
controllers/family.js

#### How should this be manually tested?
Create an estudio, a family and a member

#### Any background context you want to provide?
Faced a lot of issues while using EmbeddedDocuments, so things should change to a Document.

This template was adapted ~retrived~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: express-app/controllers/period.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const Periodo = require('../models/periodo');
+
+//options for values 
+const options = {
+  Diario: { factor: 30 , multiply: true },
+  Semanal: { factor: 4, multiply: true },
```
Comment (erikiado) [inline express-app/controllers/period.js]: 4.3?

File: express-app/controllers/transaction.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  */   
+  showTransactionView: function(request, response) {
+    //obtain familyId
+    const familyId = request.session.familyId;
+    //obtain incomes amount
+    const obtainIncomesAmount = this.getTransactionsAmount(familyId, true);
+    const obtainOutcomesAmount = this.getTransactionsAmount(familyId, false);
+    const myIncomes = this.getTransactions(familyId, true);
+    const myOutcomes = this.getTransactions(familyId, false);
+    //context variables
```
Comment (erikiado) [inline express-app/controllers/transaction.js:19]: que cholo

File: express-app/models/estudio.js [hunk 3]
```diff
@@ -50,6 +50,14 @@ class Estudio extends Document {
...
       default: opcionesStatusChoices[2],
     };
+    this.fechaCreacion = {
+      type: Date,
+      default: Date.now
+    };
+     this.editedDate = {
+      type: Date,
+      default: Date.now
+    };
```
Comment (erikiado) [inline express-app/models/estudio.js:60]: Aqui se incluye el idApi o es en otro?

File: express-app/models/miembro.js [hunk 4]
```diff
@@ -21,6 +23,37 @@ const academicDegreeChoices = [
...
+  '',
+  'tutor',
+  'madre',
+  'padre',
+  'estudiante',
+  'hermano/a',
+  'abuelo/a',
+  'tio/a',
+];
+
```
Comment (erikiado) [inline express-app/models/miembro.js:36]: Estos son un endpoint del api
captura/api-obtener-informacion-escuelas/
captura/api-obtener-informacion-oficios/

File: express-app/models/seccion.js [hunk 5]
```diff
@@ -18,6 +18,11 @@ class Seccion extends Document {
   constructor() {
     super();
 
+    this.estudioId = {
```
Comment (erikiado) [inline express-app/models/seccion.js]: Why does the section is part of an specific estudio?
This are pulled only once and contain the structure for the sections, not the actual information

File: express-app/public/css/my.css [hunk 6]
```diff
@@ -21,7 +18,19 @@ body {
...
+
+#create-members,
+#submit-income,
+#income-button,
+#estudios-btn {
+  margin-top: 20px; 
+}
+
+#add-member-button {
+  margin-bottom: 20px;
```
Comment (erikiado) [inline express-app/public/css/my.css:33]: Isnt it better to make a single class margin_top_20 or something?

File: express-app/models/miembro.js [hunk 7]
```diff
@@ -21,6 +23,37 @@ const academicDegreeChoices = [
...
+const jobOptions = [
+  '',
+  'empleado/a',
```
Comment (erikiado) [inline express-app/models/miembro.js:39]: Change the blank job for a 'ninguno' maybe based on todays meeting

File: express-app/views/dashboard.hbs [hunk 8]
```diff
@@ -39,13 +37,14 @@
             {{# if estudios }}
               {{# each estudios }}
                 <tr class="odd gradeX {%if this.status == status_options.RECHAZADO%} danger {%endif%}">
+                  <td>{{this.familia.nombreFamilia}}</td>
                   <td>{{this.familia.calle}}, {{this.familia.colonia}}, {{this.familia.codigoPostal}}</td>
-                  <td>{{this.createdDate}}</td>
-                  {{!-- <td>{{this.editedDate}}</td> --}}
+                  <td>{{this.fechaCreacion}}</td>
```
Comment (erikiado) [inline express-app/views/dashboard.hbs:42]: This dates need formatting

File: express-app/views/login.hbs [hunk 9]
```diff
@@ -56,7 +91,40 @@
 
       </div>
     </div>
-    {{> scripts }}
+    <!-- jQuery -->
+<script src="/template/vendors/jquery/dist/jquery.min.js"></script>
```
Comment (erikiado) [inline express-app/views/login.hbs:95]: PORQUE CULERO? jajaja

File: express-app/views/partials/editMemberModal.hbs [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+      <h4 class="modal-title" id="editModalLabel">Miembro</h4>
+    </div>
+    <div class="modal-body">
+      <form id="incomeSection" method="post" action="/member/edit/{{this._id}}" data-toggle="validator">
+        <div class="row">
+          <div class="form-group col-xs-12">
+            <label for="role">Rol</label>
+            <select class="form-control" id="role" name="role" readonly>
+              {{#select this.relacion}}
+                <option value=""></option>
```
Comment (erikiado) [inline express-app/views/partials/editMemberModal.hbs:15]: maybe change this value for ninguno?

File: express-app/views/partials/editIncomeModal.hbs [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+              <label for="amount">Monto*</label>
+              <input type="number" min="0" class="form-control" value={{this.monto}} name="amount" id="amount" required>
+            </div>
+          </div>
+          <div class="row">
+            <div class="form-group col-xs-6">
+              <label for="period">Peridicidad*</label>
+              <select class="form-control" id="period" name="period" required>
+                {{#select this.periocidad.periodicidad}}
+                  <option value=""></option>
```
Comment (erikiado) [inline express-app/views/partials/editIncomeModal.hbs:27]: Ninguno for all blanks or nah?

File: test/createOutcome-test.js [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+      //save student
+      return estudiante.save();
+    })
+    .then((newStudent) => {
+      //save student object
+      student = newStudent;
+      return Transaccion.count({familyId: familyId, isIngreso: false});
+    })
+    .then((count) => {
+      console.log(count);
```
Comment (erikiado) [inline test/createOutcome-test.js]: Could check for other console logs in the test

File: test/editMembers-test.js [hunk 13]
```diff
@@ -0,0 +1,383 @@
...
+      //save student
+      return estudiante.save();
+    })
+    .then((newStudent) => {
+      //save student object
+      student = newStudent;
+      return Miembro.count({familyId: familyId});
+    })
+    .then((count) => {
+      console.log(count);
```
Comment (erikiado) [inline test/editMembers-test.js]: bblbl

File: testDB/users.db [hunk 14]
```diff
@@ -1 +1 @@
-{"username":"eugenio","password":"erikiado123","apiToken":"082be9526329ebea51d69e11a50217d93f21545d","_id":"Wy3KtJBoRl6t0IVG"}
+{"username":"eugenio","password":"$2a$10$0tYBZSGH36WRrxRRb6w6QeB/W6.MvV7XJHNe341YPrrjkb5lItCBG","apiToken":"082be9526329ebea51d69e11a50217d93f21545d","_id":"Wy3KtJBoRl6t0IVG"}
```
Comment (erikiado) [inline testDB/users.db]: qp?

**Review by erikiado — CHANGES_REQUESTED** [review]

File: express-app/models/estudio.js (hunk 3 above)
Comment (Ral15) [reply express-app/models/estudio.js:60]: @erikiado simon ahí, eso no estará en este PR, esta en lo de la api

File: express-app/models/miembro.js (hunk 7 above)
Comment (Ral15) [reply express-app/models/miembro.js:39]: @erikiado maybe only add <option>Ninguno</option>>>

File: express-app/models/seccion.js (hunk 5 above)
Comment (Ral15) [inline express-app/models/seccion.js]: @adal347 was the one who created this.

@erikiado

File: express-app/public/css/my.css (hunk 6 above)
Comment (Ral15) [reply express-app/public/css/my.css:33]: @erikiado I don't agree, if this button needs more style we can add all of them there, also, at the view, it looks cleaner with just one id and not 2.

File: express-app/views/login.hbs (hunk 9 above)
Comment (Ral15) [reply express-app/views/login.hbs:95]: a mi no me jala :( @erikiado

File: express-app/views/partials/editMemberModal.hbs (hunk 10 above)
Comment (Ral15) [reply express-app/views/partials/editMemberModal.hbs:15]: `<option value="">Ninguno</option>` is better? @erikiado

File: test/createOutcome-test.js (hunk 12 above)
Comment (Ral15) [reply test/createOutcome-test.js]: @erikiado I don't understand, what could check?

File: testDB/users.db (hunk 14 above)
Comment (Ral15) [reply testDB/users.db]: not sure, I think I need to apply .gitignore changes

File: express-app/controllers/period.js (hunk 1 above)
Comment (Ral15) [reply express-app/controllers/period.js]: @erikiado I did not changed this beacause other factors need to change to.

File: express-app/controllers/transaction.js (hunk 2 above)
Comment (Ral15) [reply express-app/controllers/transaction.js:19]: ?? lo de request.locals?? o que pex @erikiado

File: express-app/models/miembro.js (hunk 4 above)
Comment (Ral15) [reply express-app/models/miembro.js:36]: 👍

File: express-app/models/seccion.js (hunk 5 above)
Comment (Ral15) [reply express-app/models/seccion.js]: 👍

File: express-app/views/dashboard.hbs (hunk 8 above)
Comment (Ral15) [reply express-app/views/dashboard.hbs:42]: @erikiado already working on this at the new branch..

File: test/createOutcome-test.js (hunk 12 above)
Comment (Ral15) [reply test/createOutcome-test.js]: @erikiado Oh I got it, travis you mean, removed then 👍 :)

File: express-app/controllers/period.js (hunk 1 above)
Comment (Ral15) [reply express-app/controllers/period.js]: 👍

**Pull request merged**